test/powershell/Modules/Microsoft.WSMan.Management/ConfigProvider.Tests.ps1

```diff
@@ -0,0 +1,478 @@
+            "WSMan:\$name" | Should Exist
+            (Get-ChildItem wsman:\).Count | Should Be ($connections.Count + 1)
+            # not a .Net type so can't use BeOfType
+            $newItem.PSObject.TypeNames[0] | Should Be "Microsoft.WSMan.Management.WSManConfigContainerElement#ComputerLevel"
+            $newItem.Name | Should Be $expected
+            Remove-Item WSMan:\$name
+            "WSMan:\$name" | Should Not Exist
+        }
+
+        It "New-Item and Remove-Item for a listener" {
+            try {
+                $newListener = New-Item -Path WSMan:\localhost\Listener\ -Address IP:127.0.0.1 -port 6666 -Transport HTTP -Enabled $false -HostName foo -URLPrefix bar -Force
+                $listenerName = $newListener.Name
+                $listenerName | Should Not BeNullOrEmpty
+                $properties = Get-ChildItem "WSMan:\localhost\Listener\$listenerName"
+                $listenerXml = [xml](winrm g winrm/config/listener?Address=IP:127.0.0.1+Transport=HTTP -format:xml)
+                $expectedMissingAttributes = "cfg","xsi","lang"
+                $properties.Count | Should Be (($listenerXml.Listener | Get-Member -Type Properties).Count - $expectedMissingAttributes.Count)
+                foreach ($property in $properties) {
+                    if (-not $property.Name.StartsWith("ListeningOn")) { # this property is represented differently
+                        $property.Value | Should Be $listenerXml.Listener.$($property.Name)
+                    }
+                }
+            }
+            finally {
+                Remove-Item -Path "WSMan:\localhost\Listener\$listenerName" -Force
+                $newListener.PSPath | Should Not Exist
+            }
+        }
+
+        It "New-Item and Remove-Item for a plugin using parameters" {
+            try {
+                $password = ConvertTo-SecureString $testPass -AsPlainText -Force
+                $creds = [pscredential]::new($testUser, $password)
+                $plugin = New-Item -Plugin TestPlugin2 -UseSharedProcess -AutoRestart `
+                    -ProcessIdleTimeoutSec 120 -FileName ${env:windir}\system32\WindowsPowerShell\v1.0\pwrshsip.dll `
+                    -SDKVersion 2 -Resource foo -Capability shell -XMLRenderingType text -Path WSMan:\localhost\plugin `
+                    -RunAsCredential $creds
+                $expectedMissingProperties = @("InitializationParameters")
+                Test-Plugin -Plugin $plugin -expectedMissingProperties $expectedMissingProperties
+            }
+            finally {
+                Remove-Item WSMan:\localhost\Plugin\TestPlugin2\
+                "WSMan:\localhost\Plugin\TestPlugin2" | Should Not Exist
+            }
+        }
+
+        It "New-Item and Remove-Item for a plugin using a file" {
+            $fileXml = @"
+<PlugInConfiguration Name="TestPlugin2" Filename="%windir%\system32\pwrshplugin.dll" SDKVersion="2" XmlRenderingType="text" Enabled="false"
```

Comment:
This is the Windows PowerShell plugin path. It matches the config you have been using throughout, but it will need to be updated

Reply:
I'll update it to the one for PSCore6